src/app/containers/InlineLink/index.new.jsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import InlineLink from '../../components/InlineLink';
+import Blocks from '../Blocks';
+import fragment from '../Fragment';
+import { inlineLinkModelPropTypes } from '../../models/propTypes/inlineLink';
+
+const InternalInlineLink = InlineLink.withComponent(Link);
+
+const componentsToRender = { fragment };
+
+const InlineLinkContainer = ({ locator, blocks }) => {
+  const InlineLinkComponent = {
+    Component: InlineLink,
+    props: { href: locator },
+  };
+
+  const regex = '^/news/articles/c[a-zA-Z0-9]{10}o$';
+  // if URL matches a valid route, use a react-router link
+  if (locator.match(regex)) {
+    InlineLinkComponent.Component = InternalInlineLink;
+    InlineLinkComponent.props = { to: locator };
+  }
+
+  // else return a normal hyperlink
```

Comment:
This comment feels misleading as it isn't actually an else and the following lines are always returned. Could we change the comments so the initial definition of `InlineLinkComponent` has a comment stating it's setup to be an absolute URL. Then the above the regex if we could have a comment mentioning the override to using a react-router link

Reply:
This file is meant to be a copy-paste of the other InlineLink component but I changed the logic slightly. I will revert to the existing logic for consistency.

Reply:
I've reverted to use the existing logic of the old InlineLink component, for consistency, @pjlee11

Reply:
Thanks, happy for the reverted changes to be done in a follow up PR so there is a single return statement and more relevant comments